### PR TITLE
Fix issues in tutorial in linux and add some of splitercontainer feature

### PR DESCRIPTION
**Pull request by MuhammadMouradG**
## Describe your changes in detail
This pull request intend to solve all problem that I encounter when I am trying with tutorials in linux.
### The changes include the following:
1. Fix directions in ScrollContainer widget.
2. Fix the size of Box widget.
3. Add parameters and documntation to SplitContainer that enhancing its behavior in gtk.
4. Replacing deprecated methods and widget like: Gtk.VBox and Gtk.VPaned.
5. Fixing icons with a relative path did not appear in the linux.
5. Fix typos.
6. Replacing `tutorial-2.png` figure with more attractive one that show the exact behavior.
## What problem does this change solve? 
In general, It fix all the problems that I encounter in the tutorials in linux.
## If this PR relates to an issue, include Refs #XXX
I am does not reversion the related issues.
screenshot:
tutorial-2
## PR Checklist:

- [] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

File: src/core/toga/command.py [hunk 1]
```diff
@@ -106,7 +108,7 @@ def icon(self, icon_or_name):
         if isinstance(icon_or_name, Icon) or icon_or_name is None:
             self._icon = icon_or_name
         else:
-            self._icon = Icon(icon_or_name)
+            self._icon = Icon(os.getcwd() + "/" + icon_or_name)
```
Comment (freakboy3742) [inline src/core/toga/command.py:111]: This may fix a bug you're seeing, but it's the wrong approach in this case. 

The name passed to `Icon` is an "icon identifier", not a filename. A filename *can* act as an icon identifier, but that shouldn't be the only interpretation. The icon identifier encompasses finding the right file extension for a given platform (.e.g, `.ico` vs `.icns`), as well as any size prefix that is required (e.g., GTK allows for icons at multiple sizes, so the icon `icons/brutus` might be rendered using `brutus-32.png` or `brutus-72.png`, depending on circumstances).

The name itself is interpreted relative to the *app* folder - i.e., the folder that declares the `__main__` entry point. so - the `icons/brutus` in tutorial 2 should be sufficient to find the icons folder in that app. The only exception to this is if you pass in the `system` flag when constructing the Icon; if `system=True`, the path is relative to the toga-core module. This means `Icon('resources/toga', system=True)` should find the Toga "yak" icon.

So - in this case, if icons aren't loading, the problem is with GTK icon path resolution, not the path being used constructing the icon. As a guide - the macOS implementation correctly finds the icons *without* the change you've proposed here.

File: src/core/toga/widgets/splitcontainer.py [hunk 2]
```diff
@@ -11,7 +11,15 @@ class SplitContainer(Widget):
...
-        content(``list`` of :class:`toga.Widget`): The list of components to be split.
+        content(``list`` of :class:`toga.Widget`): The list of components to be
+            split or tuples of components to be split and adjusting parameters
+            in the following order:
+        widget (:class:`toga.Widget`): The widget that will be added.
+        weight (float):
+        resize (boolean): Should the content expand when the widget is resized.
+        shrink (boolean): Can the content be made smaller than its requisition.
+        wide_handle (boolean): Does the seperation between the contents provide
+            stronger visual separation or notto be split.
```
Comment (freakboy3742) [inline src/core/toga/widgets/splitcontainer.py:22]: I don't think this needs to be an option. I'm OK with Toga being a little opinionated here, and picking either a wide or narrow style for split separators (and I think wide is almost always going to be preferable).

File: src/core/toga/widgets/splitcontainer.py [hunk 3]
```diff
@@ -11,7 +11,15 @@ class SplitContainer(Widget):
...
         direction: The direction for the container split,
             either `SplitContainer.HORIZONTAL` or `SplitContainer.VERTICAL`
-        content(``list`` of :class:`toga.Widget`): The list of components to be split.
+        content(``list`` of :class:`toga.Widget`): The list of components to be
+            split or tuples of components to be split and adjusting parameters
+            in the following order:
+        widget (:class:`toga.Widget`): The widget that will be added.
+        weight (float):
+        resize (boolean): Should the content expand when the widget is resized.
+        shrink (boolean): Can the content be made smaller than its requisition.
```
Comment (freakboy3742) [inline src/core/toga/widgets/splitcontainer.py:20]: I don't think this option needs to exist. If a widget sets a minimum size, that's the widget's minimum size, and a split panel shouldn't be able to override it.

File: src/core/toga/widgets/splitcontainer.py [hunk 4]
```diff
@@ -11,7 +11,15 @@ class SplitContainer(Widget):
             If no style is provided then a new one will be created for the widget.
         direction: The direction for the container split,
             either `SplitContainer.HORIZONTAL` or `SplitContainer.VERTICAL`
-        content(``list`` of :class:`toga.Widget`): The list of components to be split.
+        content(``list`` of :class:`toga.Widget`): The list of components to be
+            split or tuples of components to be split and adjusting parameters
+            in the following order:
+        widget (:class:`toga.Widget`): The widget that will be added.
+        weight (float):
+        resize (boolean): Should the content expand when the widget is resized.
```
Comment (freakboy3742) [inline src/core/toga/widgets/splitcontainer.py:19]: For consistency in terminology, this should be "flex" rather than "resize".

File: src/core/toga/widgets/splitcontainer.py [hunk 5]
```diff
@@ -55,17 +63,26 @@ def content(self, content):
         self._content = []
         for position, item in enumerate(content):
             if isinstance(item, tuple):
-                widget, weight = item
+                if len(item) == 5:
```
Comment (freakboy3742) [inline src/core/toga/widgets/splitcontainer.py:66]: This will break any existing code using weighted splits. I would suggest that we should accomodate tuples of length 2 *and* 3 (3, rather than 5, because the shrink and wide_handle options aren't required). Tuples of length 2 would default to `flex=True`.

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
Thanks for the contribution! That box scrolling problem has been lurking for a while, so thanks for getting to the bottom of it.

On the whole, the code looks good; I've flagged a couple of problems inline. More generally, this might have been handled better as 2 PRs - one fixing the box size problem for scrolling, and one adding new features to the split pane. There's no need to split this PR into two; but generally speaking, it's better to keep each PR addressing one specific feature or bug, rather than a generic "fix the tutorial" that encompasses multiple problems. Smaller PRs are much easier to review as there are less interactions to consider, and less complexity means faster reviews.

File: src/core/toga/widgets/splitcontainer.py [hunk 6]
```diff
@@ -55,17 +60,22 @@ def content(self, content):
         self._content = []
         for position, item in enumerate(content):
             if isinstance(item, tuple):
-                widget, weight = item
+                if len(item) == 2:
+                    widget, weight = item
+                    flex = True
+                elif len(item) == 3:
+                    widget, weight, flex = item
```
Comment (freakboy3742) [inline src/core/toga/widgets/splitcontainer.py:67]: What if the length of the tuple isn't 2 or 3?

**Review by freakboy3742 — CHANGES_REQUESTED** [review]
This is starting to look good; one edge case of error handling flagged inline.

Tests are also failing; I'm guessing from the failure that is being reported, you'll also start seeing failures in the Cocoa and Winforms backend tests, because the prototype for add_content has changed.

**Review by MuhammadMouradG — COMMENTED** [review]
Does that mean that I can edit the tests to correct that failure?

**Comment by freakboy3742** [comment]
The tests need to pass before I can merge anything - so yes, you'll need to update the code until the tests pass.

That might involve modifying the *test* - or adding *new* tests (in this case, since there are new ways to invoke SplitContainer, new tests would seem appropriate). 

However, in this case, you'll also need to update the `dummy` backend. You've added an argument to the `add_content` method. That method is part of the interface contract for backends, but the `dummy` backend that is used for tests doesn't include that argument. So, you'll need to modify the `dummy` backend's implementation of SplitContainer to accept and handle the new argument that you've introduced.

File: src/core/toga/command.py (hunk 1 above)
Comment (MuhammadMouradG) [reply src/core/toga/command.py]: In regard to icons, as I was browsing through the code, I found that the code depends on finding the location of the images on several things, which including that the `helloworld` in command `python -m helloworld` must be a folder not a `Python` file, `helloworld` treated as a module by using `-m` flag when executing the program, and `importlib_metadata.metadata` is defined, otherwise `Tog` will be automatically called my `app_name` and also the `module_name` as `toga` rather than `helloworld` and this will miss the execution the path to icons folder.

But because `Toga` is primarily treated as part of `BeeWare` project, which implicitly define the previous parameters correctly while creating the app, I tried the second lesson of `Toga` with `BeeWare` and succeeded in finding the icons. 

So, what do you think?

**Review by freakboy3742 — APPROVED** [review]
Awesome! Looks great - thanks for the contribution!
